Team manual rst2pdf error: No such file or directory: 'fc-match'

The error occurs when the team manual is generated using sphinx-build/rst2pdf in the contributor and domserver Docker containers.

The rst2pdf package uses the `fc-match` command from the fontconfig package, but doesn't list it as a dependency for some reason so fontconfig isn't installed.

The build exit status is 0 (success), but the output includes a long error message and the generated file domjudge-team-manual.pdf is empty.

(The build happens to work in the gitlabci container because it installs packaging-dev, which pulls in fontconfig.)

### Steps to reproduce

```
git clone https://github.com/DOMjudge/domjudge.git
cd domjudge
git checkout 656477610b41
docker run -v $PWD:/domjudge -it domjudge/domjudge-contributor@sha256:b4cc5d9069aac62497e39ee320b199cd31fe5bc72499fed96a5fd505990607a7
```

### Expected behaviour

domjudge/doc/manual/build/team/domjudge-team-manual.pdf should contain the team manual.

### Actual behaviour

domjudge-team-manual.pdf exists but is empty.

### Build output

<details>

```
...
make html team
make[2]: Entering directory '/domjudge/doc/manual'
...
sphinx-build -b pdf . build/team
Running Sphinx v1.6.7
making output directory...
loading translations [en]... done
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [pdf]: targets for 18 source files that are out of date
updating environment: 18 added, 0 changed, 0 removed
reading sources... [100%] upgrading
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
processing domjudge-team-manual... 
team 
resolving references...
done
writing domjudge-team-manual... 
[ERROR] pdfbuilder.py:149 [Errno 2] No such file or directory: 'fc-match': 'fc-match'
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/rst2pdf/pdfbuilder.py", line 146, in write
    docwriter.write(doctree, destination)
  File "/usr/lib/python3/dist-packages/docutils/writers/__init__.py", line 80, in write
    self.translate()
  File "/usr/local/lib/python3.6/dist-packages/rst2pdf/pdfbuilder.py", line 643, in translate
    section_header_depth=self.section_header_depth
  File "/usr/local/lib/python3.6/dist-packages/rst2pdf/createpdf.py", line 173, in __init__
    self.loadStyles(stylesheets)
  File "/usr/local/lib/python3.6/dist-packages/rst2pdf/createpdf.py", line 276, in loadStyles
    def_dpi=self.def_dpi)
  File "/usr/local/lib/python3.6/dist-packages/rst2pdf/styles.py", line 358, in __init__
    fontList = findfonts.autoEmbed(style[key])
  File "/usr/local/lib/python3.6/dist-packages/rst2pdf/findfonts.py", line 322, in autoEmbed
    variants = findTTFont(fname)
  File "/usr/local/lib/python3.6/dist-packages/rst2pdf/findfonts.py", line 214, in findTTFont
    family = get_family(fname)
  File "/usr/local/lib/python3.6/dist-packages/rst2pdf/findfonts.py", line 178, in get_family
    data = make_string(subprocess.check_output(["fc-match", query]))
  File "/usr/lib/python3.6/subprocess.py", line 356, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.6/subprocess.py", line 423, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.6/subprocess.py", line 729, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.6/subprocess.py", line 1364, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'fc-match': 'fc-match'
FAILED
build succeeded.
make[2]: Leaving directory '/domjudge/doc/manual'
...
```

</details>

### Fix

The obvious fix is to install fontconfig. There may be other solutions though, I'm not familiar with rst2pdf.

### Pull request

#57, DOMjudge/domjudge#797 (for install instructions)

--- a/debian/control
+++ b/debian/control
@@ -5,7 +5,7 @@ Maintainer: DOMjudge Developers <team@domjudge.org>
 Uploaders: Jaap Eldering <jaap@jaapeldering.nl>, Keith Johnson <kj@ubergeek42.com>, Thijs Kinkhorst <thijs@debian.org>
 Build-Depends: libcurl4-gnutls-dev, libmagic-dev,
  debhelper (>= 12), libcgroup-dev, zip, libjsoncpp-dev
-Build-Depends-Indep: python-sphinx, python-sphinx-rtd-theme, rst2pdf
+Build-Depends-Indep: python-sphinx, python-sphinx-rtd-theme, rst2pdf, fontconfig
 Homepage: https://www.domjudge.org
 Standards-Version: 4.5.0
 Vcs-Git: https://github.com/DOMjudge/domjudge-packaging.git

--- a/docker-contributor/Dockerfile
+++ b/docker-contributor/Dockerfile
@@ -22,7 +22,7 @@ RUN apt-get update \
     php7.2-gmp php7.2-xml php7.2-mbstring php-xdebug \
     bsdmainutils ntp \
     linuxdoc-tools linuxdoc-tools-text groff \
-    python3-sphinx python3-sphinx-rtd-theme python3-yaml python3-pip \
+    python3-sphinx python3-sphinx-rtd-theme python3-pip fontconfig python3-yaml \
     texlive-latex-recommended texlive-latex-extra \
     texlive-fonts-recommended texlive-lang-european \
     sudo debootstrap libcgroup-dev procps \
@@ -31,7 +31,6 @@ RUN apt-get update \
     supervisor apache2-utils lsb-release \
     libcurl4-gnutls-dev libjsoncpp-dev libmagic-dev python python-pip \
     enscript lpr ca-certificates less vim \
-    && pip install Pygments \
     && rm -rf /var/lib/apt/lists/*
 
 # Needed for building the docs

--- a/docker-gitlabci/Dockerfile
+++ b/docker-gitlabci/Dockerfile
@@ -7,7 +7,7 @@ RUN apt-get update && apt-get install -y \
   devscripts shellcheck nginx libboost-regex-dev \
   php php-cli php-gd php-curl php-mysql php-json php-gmp php-zip php-xml php-mbstring php-fpm php-intl \
   # Docs \
-  python-sphinx python-sphinx-rtd-theme rst2pdf python3-yaml \
+  python-sphinx python-sphinx-rtd-theme rst2pdf fontconfig python3-yaml \
   texlive-latex-recommended texlive-latex-extra \
   texlive-fonts-recommended texlive-lang-european \
   # Misc gitlab things \

--- a/docker/domserver/Dockerfile
+++ b/docker/domserver/Dockerfile
@@ -13,7 +13,7 @@ RUN apt-get update \
 	php-gmp php-xml php-mbstring \
 	sudo bsdmainutils ntp libcgroup-dev procps \
 	linuxdoc-tools linuxdoc-tools-text groff \
-	python3-sphinx python3-sphinx-rtd-theme python3-yaml python3-pip \
+	python3-sphinx python3-sphinx-rtd-theme python3-pip fontconfig python3-yaml \
 	texlive-latex-recommended texlive-latex-extra \
 	texlive-fonts-recommended texlive-lang-european \
 	libcurl4-gnutls-dev libjsoncpp-dev libmagic-dev \

